packages/flutter_adaptive_scaffold/test/adaptive_scaffold_test.dart

```diff
@@ -127,6 +127,30 @@ void main() {
     expect(tester.getBottomRight(sBody), const Offset(800, 800));
   });
 
+  // Regression test for https://github.com/flutter/flutter/issues/111008
+  testWidgets(
+    'appBar parameter should have the type PreferredSizeWidget',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(size: Size(500, 800)),
+          child: AdaptiveScaffold(
+            drawerBreakpoint: TestBreakpoint0(),
+            internalAnimations: false,
+            destinations: const <NavigationDestination>[
+              NavigationDestination(icon: Icon(Icons.inbox), label: 'Inbox'),
+              NavigationDestination(
+                  icon: Icon(Icons.video_call), label: 'Video'),
+            ],
+            appBar: AppBar(),
```

Comment:
When using a `AppBar` widget, the test would still pass when in v0.0.5 and before. So technically, this test isn't a regression test for https://github.com/flutter/flutter/issues/111008, right? This was the reason why I created an empty `PreferredSizeWidget` in #2617 so that it would fail when using `AppBar` as a type of `appBar` parameter.

Reply:
https://github.com/flutter/flutter/blob/23f22cb8535c0e8746badef943c8c178d30b5f07/packages/flutter/lib/src/material/app_bar.dart#L170
```dart
class AppBar extends StatefulWidget implements PreferredSizeWidget
```

From https://github.com/flutter/packages/pull/2617/files#diff-7fb7fb81b4dd090876718f62145eee821681e35d653653292dba23956c941c30R152-R153
```dart
class PreferredSizeWidgetImpl extends StatelessWidget implements PreferredSizeWidget
```